### PR TITLE
Add more profile GUID tests

### LINKED ISSUE #2119: Add some tests for invalid GUIDs in JSON
> 
> 
> Should we add tests for the following?
> 
>     * [ ]  GUID in different GUID format
> 
>     * [ ]  completely invalid GUID (i.e.: guid = )
> 
>     * [ ]  GUID with letters in caps
> 
> 
> Or would the first two just be examples of invalid settings values anyways so there's no point?

From @carlos-zamora in #2117

### PR DESCRIPTION
**Pull request by marcelwgn**
## Summary of the Pull Request
Adding more profile GUID tests
## References and Relevant Issues
Closes #2119 
## Detailed Description of the Pull Request / Additional comments

Currently, there are formats that simply break GUID parsing (the commented out test cases). Should we catch that and treat it like a "null" GUID or should we generate a new GUID for those profiles?

## PR Checklist
- [x] Closes #2119 
- [x] Tests added/passed
- [ ] Documentation updated
   - If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
- [ ] Schema updated (if necessary)

### CODE REVIEW

File: src/cascadia/UnitTests_SettingsModel/ProfileTests.cpp [hunk 1]
```diff
@@ -55,37 +64,50 @@ namespace SettingsModelUnitTests
...
-        const winrt::guid nullGuid{};
+        parseAndVerifyProfile(profileWithoutGuid, false);
+        parseAndVerifyProfile(secondProfileWithoutGuid, false);
 
-        VERIFY_IS_FALSE(profile0->HasGuid());
-        VERIFY_IS_FALSE(profile1->HasGuid());
-        VERIFY_IS_FALSE(profile2->HasGuid());
-        VERIFY_IS_TRUE(profile3->HasGuid());
-        VERIFY_IS_TRUE(profile4->HasGuid());
+        // The following crash JSON parsing
```
Comment (DHowett) [inline src/cascadia/UnitTests_SettingsModel/ProfileTests.cpp:96]: FWIW, this should throw an exception which you can verify you caught with `VERIFY_THROWS`!

Comment (marcelwgn) [reply src/cascadia/UnitTests_SettingsModel/ProfileTests.cpp:96]: I wasn't sure if they should crash so I left them as comment, now they are using VERIFY_THROWS (thanks for mentioning that!)

**Review by DHowett — APPROVED** [review]
thanks so much!

**Review by carlos-zamora — APPROVED** [review]
This is great! Thanks!

**Pull request merged**